README.md

```diff
@@ -438,7 +438,8 @@ itself.
 
 ### Event: `'drain'`
 
-A `'drain'` event is emitted whenever the `queueSize` reaches `0`.
+A `'drain'` event is emitted whenever the `queueSize` is below the
```

Comment:
I'm not sure we should do it. The docs clearly show an example:

> The 'drain' event may be used to receive notification when the queue is empty and all tasks have been submitted to workers for processing. -- https://github.com/piscinajs/piscina#backpressure

Reply:
That's a good point, and the current approach might not be the more optimal as it requires waiting until the queue is totally empty before resuming work, which can not be the best depending on the workload (e.g. extending waiting times, etc.)

The idea was to allow the pool to notify whenever there is spare capacity to be used for scheduling new tasks, maybe we can think of something better?

Reply:
IMHo this should work the same as `stream.Writable` or be called something else. Otherwise it confusing.

Reply:
If we change this behaviour, it should be a semver-major.

@ronag I couldn't find it in the docs, but the behaviour stream.Writable does it the same as this PR is trying to apply?

Reply:
>If we change this behaviour, it should be a semver-major.

Sure, is already marked as so and pointing to `next` branch 👍 

>@ronag I couldn't find it in the docs, but the behaviour stream.Writable does it the same as this PR is trying to apply?

Yeah, have the same issue when trying to check `stream.Writable`, so how we should move forward? 🙂

Reply:
@mcollina @jasnell any thoughts on this?

Reply:
This description needs to be updated with the current intended behaviour.